design-proposals/vm-maintenance-mode.md

```diff
@@ -0,0 +1,61 @@
+# Overview
+Add a new "Maintenance" phase for the VMI/VM that represents a special state where the VM doesn't run and we can safely access the VM disks.
+
+![VMI phase with Maintenance](vm-maintenance-mode-images/VMI_phase.png)
+![VM phase with Maintenance](vm-maintenance-mode-images/VM_phase.png)
+
+The "maintenance" pods are pods created to perform disk operations (like for example to modify the disk with libguestfs-tools). While the VM is in this state, we can safely create maintenance pods that can access VM disks. The maintenance pods must have the same scheduling constraints as the VM in oder to not alterate the scheduling of the VM.
+
+The new Maintenance state differs from Stopped because in this state we refuse to start the VM. Leaving the Maintenance state needs to be an explicit action. We can move the VM/VMI from the Maintenance state back to its previous state only in the case there are not active maintenance pods.
```

Comment:
Might make sense to state that the only valid state transitions involving 'Maintenance' are:
* Stopped --> Maintenance
* Maintenance --> Stopped